packages/api/src/objectSet/ObjectSet.ts

```diff
@@ -385,13 +607,13 @@ interface Subscribe<
 interface ObjectSetCleanedTypes<
   Q extends ObjectOrInterfaceDefinition,
   D extends Record<string, SimplePropertyDef>,
-  MERGED extends ObjectOrInterfaceDefinition,
+  MERGED extends ObjectOrInterfaceDefinition & Q,
 > extends
   MinimalObjectSet<Q, D>,
   WithProperties<Q, D>,
   Aggregate<MERGED>,
   SetArithmetic<MERGED>,
-  PivotTo<MERGED>,
+  PivotTo<Q>,
```

Comment:
I don't think it makes sense to retain the "merged" object when you pivot since those properties are contextually gone now.